web-console/src/dialogs/doctor-dialog/doctor-checks.tsx

```diff
@@ -0,0 +1,421 @@
+import { pluralIfNeeded, queryDruidSql } from '../../utils';
+import { deepGet } from '../../utils/object-change';
+import { postToSampler } from '../../utils/sampler';
+
+export interface CheckControls {
+  addSuggestion: (message: string) => void;
+  addIssue: (message: string) => void;
+  terminateChecks: () => void;
+}
+
+export interface DoctorCheck {
+  name: string;
+  check: (controls: CheckControls) => Promise<void>;
+}
+
+const RUNTIME_PROPERTIES_ALL_NODES_MUST_AGREE_ON: string[] = [
+  'user.timezone',
+  'druid.zk.service.host',
+];
+
+const RUNTIME_PROPERTIES_ALL_NODES_SHOULD_AGREE_ON: string[] = ['java.version'];
+
+// In the future (when we can query other nodes) is will also be cool to check:
+// 'druid.storage.type' <=> historicals, overlords, mm
+// 'druid.indexer.logs.type' <=> overlord, mm, + peons
+
+const RUNTIME_PROPERTIES_MASTER_NODES_SHOULD_AGREE_ON: string[] = [
+  'druid.metadata.storage.type', // overlord + coordinator
+  'druid.metadata.storage.connector.connectURI',
+];
+
+export const DOCTOR_CHECKS: DoctorCheck[] = [
+  // -------------------------------------
+  // Self (router) checks
+  // -------------------------------------
+  {
+    name: 'Verify own status',
+    check: async controls => {
+      // Make sure that the router responds to /status and gives some valid info back
+      let status: any;
+      try {
+        status = (await axios.get(`/status`)).data;
+      } catch (e) {
+        controls.addIssue(
+          `Did not get a /status response, is the cluster running? Got: ${e.message}`,
+        );
+        controls.terminateChecks();
+        return;
+      }
+
+      if (typeof status.version !== 'string') {
+        controls.addIssue('Could not get a valid /status response.');
+      }
+    },
+  },
+  {
+    name: 'Verify own runtime properties',
+    check: async controls => {
+      // Make sure that everything in /status/properties is above board
+      let properties: Record<string, string>;
+      try {
+        properties = (await axios.get(`/status/properties`)).data;
+      } catch (e) {
+        controls.addIssue('Did not get a /status/properties response, something must be broken.');
+        return;
+      }
+
+      // Check that the management proxy is on, it really should be for someone to access the console in the first place but everything could happen
+      if (properties['druid.router.managementProxy.enabled'] !== 'true') {
+        controls.addIssue(
+          `The router's "druid.router.managementProxy.enabled" is not reported as "true" that is unusual.`,
+        );
+      }
+
+      // Check that the underlying Java is Java 8 the only officially supported Java version at the moment.
+      if (
+        properties['java.runtime.version'] &&
+        !properties['java.runtime.version'].startsWith('1.8')
+      ) {
+        controls.addSuggestion(
+          `It looks like are running Java ${properties['java.runtime.version']}, Druid only officially supports Java 1.8.x`,
```

Comment:
`java.specification.version` may be easier to parse. (startsWith 1.8 is a little brittle, what happens when 1.80 is out 🙂)

A period seems more grammatical to me here (vs. a comma).